### PR TITLE
Convert erb templates to haml (4)

### PR DESCRIPTION
**Pull request by eduardoj**
Replace ERB templates with HAML for the rest of the views in shared/.

### CODE REVIEW

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      %li
+        %a{class: "result_select_link_#{index}", href: '#', id: "result_select_link_#{index}_1"} Rpmlint Results
+        = sprite_tag('reload', title: 'Reload', onclick: 'update_rpmlint_result()', class: "result_reload_#{index}", id: "result_reload_#{index}_1", style: 'display: none')
+        = image_tag('ajax-loader.gif', id: "result_spinner_#{index}_1", class: 'hidden')
+%div{class: "result_display_#{index}", id: "result_display_#{index}_0"}
+- if controller == 'package'
+  %div{class: "result_display_#{index}", id: "result_display_#{index}_1"}
+= javascript_tag do
+  - if controller == 'package'
+    $('.result_select_link_#{index}').click(function (event) {
```
Comment (DavidKang) [inline src/api/app/views/shared/_buildresult_box.html.haml]: It's bad indented

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    \#{raw("'project': '#{project}',") if defined?(project)}
+    \#{raw("'package': '#{package}',") if defined?(package)}
+    \#{raw("'index': '#{index}',") if defined?(index)}
+    },
+    success: function(data) {
+    $('#result_display_#{index}_1').html(data);
+    },
+    error: function(data) {
+    $('#result_display_#{index}_1').html('
+    %p> No rpmlint results available
```
Comment (DavidKang) [inline src/api/app/views/shared/_buildresult_box.html.haml]: strange tag `%p>` :smile_cat:

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  \#{raw("'project': '#{project}',") if defined?(project)}
+  \#{raw("'package': '#{package}',") if defined?(package)}
+  \#{raw("'index': '#{index}',") if defined?(index)}
+  },
+  success: function(data) {
+  $('#result_display_#{index}_0').html(data);
+  },
+  error: function(data) {
+  $('#result_display_#{index}_0').html('
+  %p> No build results available
```
Comment (DavidKang) [inline src/api/app/views/shared/_buildresult_box.html.haml]: strange tag `%p>` :smile_cat:

File: src/api/app/views/shared/_editor.html.haml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+- uid ||= next_codemirror_uid
+- content_for(:head_style, codemirror_style(style))
+- unless style[:read_only]
+  %div{ id: "top_#{uid}" }
+    .toolbar
+      %span{ style: 'float: left' }
+        %input.tools.buttons.small.save.inactive{ id: "save_#{uid}", type: 'button', value: 'Save' }/
+        %input.tools.buttons.undo{ id: "undo_#{uid}", onclick: "editors[#{uid}].Undo(this);", style: 'background: transparent', type: 'button', value: 'undo' }/
+        %input.tools.buttons.redo{ id: "redo_#{uid}", onclick: "editors[#{uid}].Redo(this);", style: 'background: transparent', type: 'button', value: 'redo' }/
+        /
```
Comment (DavidKang) [inline src/api/app/views/shared/_editor.html.haml]: [nitpick] Could you remove this commented code?

**Comment by Ana06** [comment]
@eduardoj did you corrected everything automatically or you also did manual changes?

**Comment by eduardoj** [comment]
> @eduardoj did you corrected everything automatically or you also did manual changes?

I corrected everything automatically and also did manual changes.

**Comment by Ana06** [comment]
@eduardoj 
> I corrected everything automatically and also did manual changes.

Would be nice that next time you separate this changes in different commits, it is easier to review and to identity were possible error come from. :wink:

**Comment by evanrolfe** [comment]
> Would be nice that next time you separate this changes in different commits, it is easier to review and to identity were possible error come from.

I agree with this in general. Think it would be better if we did not use `git commit --amend` but made our changes in seperate commits and then squashed them right before merging.

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      update_rpmlint_result();
+  :plain
+
+    function update_build_result() {
+      $('#result_spinner_#{index}_0').show();
+      $('#result_reload_#{index}_0').hide();
+      $.ajax({
+        url: '#{url_for(controller: controller, action: 'buildresult')}',
+        data: {
+          \#{raw("'project': '#{project.to_param}',") if defined?(project)}
```
Comment (Ana06) [inline src/api/app/views/shared/_buildresult_box.html.haml]: spaces around brackets:

`\#{raw("'project': '#{ project }',") if defined?(project)}`

and also in the next 3 lines.

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        return false;
+      });
+
+      function update_rpmlint_result() {
+        $('#result_spinner_#{index}_1').show();
+        $('#result_reload_#{index}_1').hide();
+        $.ajax({
+          url: '#{url_for(controller: controller, action: 'rpmlint_result')}',
+          data: {
+            \#{raw("'project': '#{project.to_param}',") if defined?(project)}
```
Comment (Ana06) [inline src/api/app/views/shared/_buildresult_box.html.haml]: same here

File: src/api/app/views/shared/_buildresult_box.html.haml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      = image_tag('ajax-loader.gif', id: "result_spinner_#{index}_0")
+    - if controller == 'package'
+      %li
+        %a{class: "result_select_link_#{index}", href: '#', id: "result_select_link_#{index}_1"} Rpmlint Results
+        = sprite_tag('reload', title: 'Reload', onclick: 'update_rpmlint_result()', class: "result_reload_#{index}", id: "result_reload_#{index}_1", style: 'display: none')
+        = image_tag('ajax-loader.gif', id: "result_spinner_#{index}_1", class: 'hidden')
+%div{class: "result_display_#{index}", id: "result_display_#{index}_0"}
+- if controller == 'package'
+  %div{class: "result_display_#{index}", id: "result_display_#{index}_1"}
+= javascript_tag do
```
Comment (Ana06) [inline src/api/app/views/shared/_buildresult_box.html.haml]: leave an empty line before the javascript please

File: src/api/app/views/shared/_sourcediff.html.haml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                %a{ href: '#', id: "expand_#{file_id}", onclick: "collapse_expand(\'#{file_id}\'); return false;" } [+]
+                - content_for :ready_function do
+                  - if is_expanded
+                    $('#expand_#{file_id}').hide();
+                  - else
+                    $('#collapse_#{file_id}').hide();
+            %td{ style: 'width: 1%' }= file_state
+            %td
+              /
+                Most files don't exist when the package is linked, so keeping in mind the statement below don't even bother
```
Comment (Ana06) [inline src/api/app/views/shared/_sourcediff.html.haml]: this comment can be shorted and written in a nicer way, removing for example the `...`

**Comment by eduardoj** [comment]
> Some of the commits can be squashed. 😉

Done. Created a "Minor code style changes" commit with all of them. @Ana06 please re-review.

**Pull request merged**